moto/resourcegroups/models.py

```diff
@@ -368,6 +383,37 @@ def put_group_configuration(self, group_name, configuration):
         self.groups.by_name[group_name].configuration = configuration
         return self.groups.by_name[group_name]
 
+    def group_resources(self, group_name, resource_arns):
+        self.groups.by_name[group_name].resource_arns = resource_arns
+
+    def ungroup_resources(self, group_name, resource_arns):
+        actual_resource_arns = self.groups.by_name[group_name].resource_arns
+        for resource_arn in resource_arns:
+            actual_resource_arns.remove(resource_arn)
+        self.groups.by_name[group_name].resource_arns = actual_resource_arns
+
+    def list_group_resources(self, group_name):
+        return self.groups.by_name[group_name].resource_arns
+
+    def search_resources(self, resource_query):
```

Comment:
This method should have a few tests as well, covering all possible use cases (no matches, 1 resource-type match, 1 tag match, and combination of the above)